### ISSUE TITLE
False positive "main function not found" with rocket

### ISSUE THREAD

**Issue by Undin** [intellij-rust/intellij-rust#5975]
## Environment

* **IntelliJ Rust plugin version:** 0.3.130.3298-202-nightly
* **Rust toolchain version:** 1.47.0-nightly (9b88e0a86 2020-08-15) x86_64-apple-darwin
* **IDE name and version:** IntelliJ IDEA 2020.2.1 Preview Ultimate Edition (IU-202.6948.5)
* **Operating system:** macOS 10.13.6

## Problem description

Currently, the plugin doesn't expand procedural macros. As a result, it can't find generated `main` function and produces "`main` function not found" error although it's generated by proc macro.

![image](https://user-images.githubusercontent.com/2539310/90922229-4fe2a400-e3f4-11ea-86ec-43537e1b191e.png)

Of course, it will be automatically fixed when proc macros will be supported by the plugin. But, probably, we can check that crate root contains proc macro invocations (via direct call or via attributes) and disable the corresponding inspection til proc macros aren't supported by the plugin

## Steps to reproduce
```rust
#[macro_use] extern crate rocket;

#[cfg(test)] mod tests;

#[get("/")]
fn hello() -> &'static str {
    "Hello, world!"
}

#[launch]
fn rocket() -> rocket::Rocket {
    rocket::ignite().mount("/", routes![hello])
}

```
Code is taken from [rocket](https://github.com/SergioBenitez/Rocket/blob/02b79e28f71ccbd1fce0e369469c92caedb38cd8/examples/hello_world/src/main.rs) repo

**Comment by AwesomestCode** [comment]
Same issue happening with me; can confirm.

**Comment by ngn999** [comment]
same issue in uu_coreutils repo: https://github.com/uutils/coreutils.

**Comment by miathedev** [comment]
I can confirm this issue too. It confused me a bit.

**Comment by JoshuaNitschke** [comment]
Me too

**Comment by poovamraj** [comment]
A small workaround this would be something like this

```rust
#[macro_use] extern crate rocket;

#[get("/")]
fn index() -> &'static str {
    "Hello, world!"
}

#[rocket::main]
async fn main() {
    rocket::build().mount("/", routes![index]).launch().await;
}
```

**Comment by niklasfi** [comment]
@poovamraj good suggestion, it leads to a warning though:

```
warning: unused `Result` that must be used
  --> src/main.rs:10:5
   |
10 |     rocket::build().mount("/", routes![index]).launch().await;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: `#[warn(unused_must_use)]` on by default
   = note: this `Result` may be an `Err` variant, which should be handled
```

which can be disabled using `#[allow(unused_must_use)]`:

```rust
#[macro_use] extern crate rocket;

#[get("/")]
fn index() -> &'static str {
    "Hello World!"
}

#[rocket::main]
#[allow(unused_must_use)]
async fn main() {
    rocket::build().mount("/", routes![index]).launch().await;
}
```

**Comment by nanoqsh** [comment]
@niklasfi
Or you can just "use" this
```rust
let _ = rocket::build().mount("/", routes![index]).launch().await;
```

**Comment by splaa** [comment]
#[rocket::main]
async fn main() {
    let _ = rocket::build().mount("/", routes![index]).launch().await.expect("TODO: panic message");
}

**Comment by tmvkrpxl0** [comment]
Enabling proc macro as stated on #6908 on experimental setting still does not solve the issue

**Comment by Undin** [comment]
Fixed since today [nightly](https://plugins.jetbrains.com/plugin/8182-rust/docs/rust-quick-start.html#install-nightly) if `org.rust.macros.proc.attr` [experimental feature](https://plugins.jetbrains.com/plugin/8182-rust/docs/rust-faq.html#experimental-features) is enabled

**Comment by RenaudDenis** [comment]
> Fixed since today [nightly](https://plugins.jetbrains.com/plugin/8182-rust/docs/rust-quick-start.html#install-nightly) if `org.rust.macros.proc.attr` [experimental feature](https://plugins.jetbrains.com/plugin/8182-rust/docs/rust-faq.html#experimental-features) is enabled

No cigar for me unfortunately

**Comment by Maksim20023** [comment]
Hi @RenaudDenis! Unfortunately, there isn't enough information to investigate the causes of the issue you facing. So it would be great if you could provide me more data such as your environment (including enabled experimental features), code sample, and steps to reproduce.

**Comment by yshavit** [comment]
This works for me with actix-web (that is, I do not get the error). It didn't at first (I _did_ get the error), but after updating to the latest Intellij, it's now fine. I'm following the steps at https://actix.rs/docs/getting-started.

## src/main.rs
```rust
use actix_web::{get, App, HttpResponse, HttpServer, Responder};

#[get("/")]
async fn hello() -> impl Responder {
    HttpResponse::Ok().body("Hello world!")
}

#[actix_web::main] // error goes away if I comment this out
async fn main() -> Result<(), std::io::Error> {
    HttpServer::new(|| App::new().service(hello))
        .bind(("127.0.0.1", 8080))?
        .run()
        .await
}
```

## Cargo.toml

```toml
[package]
name = "irtest"
version = "0.1.0"
edition = "2021"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
actix-web = "4"
```

## System info
* macOS 11.7.8
* `cargo --version` => 1.71.0
* `rustc --version` => 1.71.0
* intellij version =>

**Comment by RenaudDenis** [comment]
Last time I tried it was resolved.
I supposed it has been fixed after a clion update.

**Comment by beyarkay** [comment]
It's unlikely anyone was silly like me, but I accidentally put

```rust
#![cfg(debug_assertions)]
```
at the top of my `main.rs` and forgot about it, which resulted in debug builds being fine, but `--release` builds failed with the confusing error message mentioned above. Hopefully this helps someone else out.
